### PR TITLE
tsp, add test for cadl-ranch

### PR DESCRIPTION
**Pull request by v-hongli1**
1. Add test cases for Azure/ResourceManager/Models/CommonTypes/ManagedIdentity
2. Add test cases for Type/Property/Optional/Plaindate
3. Add test cases for Type/Property/Optional/Plaintime
4. Add test cases for azure/core/model

### CODE REVIEW

File: typespec-tests/src/test/java/com/azure/resourcemanager/models/commontypes/managedidentity/ManagedIdentityManagerTests.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                        .withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED)
+                        .withUserAssignedIdentities(userAssignedIdentityMap))
+                .apply();
+        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED, resource.identity().type());
+        Assertions.assertNotNull(resource.identity().principalId());
+        Assertions.assertNotNull(resource.identity().tenantId());
+        Assertions.assertNotNull(resource.identity().userAssignedIdentities());
+        Assertions.assertEquals(1, resource.identity().userAssignedIdentities().size());
+        resource.identity().userAssignedIdentities()
+                .values().forEach(item -> {
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/azure/resourcemanager/models/commontypes/managedidentity/ManagedIdentityManagerTests.java:59]: Don't use forEach, you only have 1 item.

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/azure/resourcemanager/models/commontypes/managedidentity/ManagedIdentityManagerTests.java]: Fixed in the new version.

File: typespec-tests/src/test/java/com/azure/resourcemanager/models/resources/ArmResourceTest.java
[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply typespec-tests/src/test/java/com/azure/resourcemanager/models/resources/ArmResourceTest.java:1]: My local test is `valid2` as well.
This should be `valid2`. I think if LRO initial request returns 200, we directly use its response body as model and won't do a `get` request again.
Let me know if it's still `valid` in your local setup @v-hongli1

[the remark this answers is not in the document]
Comment (v-hongli1) [reply typespec-tests/src/test/java/com/azure/resourcemanager/models/resources/ArmResourceTest.java:1]: It's still valid in my local setup，so all changes of `ArmResourceTes`t have been rollbacked.

[the remark this answers is not in the document]
Comment (XiaofeiCao) [reply typespec-tests/src/test/java/com/azure/resourcemanager/models/resources/ArmResourceTest.java:1]: Thanks. I reproduced as well, though after using the latest cadl-ranch, mine is fixed as well..
Strange, as there's nothing different between cadl-ranch versions..

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**